src/main.js

```diff
@@ -107,11 +128,19 @@ try {
 	console.error(err);
 }
 
-// Set path according to being built or not
+// If root user, use a root-owned directory for user data
+var configDir = path.join(process.env.HOME, product.dataFolderName);
 if (process.env.VSCODE_DEV) {
-	var appData = app.getPath('appData');
-	app.setPath('userData', path.join(appData, 'Code-Development'));
+	configDir += '-dev';
 }
+var userDataDir = path.join(configDir, 'user-data');
+
+// Attempt migrations if userDataDir does not exist
+if (!fs.existsSync(userDataDir)) {
+	migrateUserDataDir(userDataDir);
```

Comment:
Would this not mean we could potentially migrate the user folder from normal user to root? I would think we should not migrate for root...